src/mosquito/queue.cr

```diff
@@ -184,12 +169,36 @@ module Mosquito
       end.uniq.flatten
     end
 
-    def ==(other : self)
+    def ==(other : self) : Bool
       name == other.name
     end
 
     def flush
       Redis.instance.del waiting_q, pending_q, scheduled_q, dead_q
     end
+
+    # Determines if a task needs to be throttled and not dequeued
+    def rate_limited? : Bool
+      # Get the latest config for the queue
+      @config = get_config
+
+      # Return if throttleing is not needed
+      return false if @config["limit"].to_i.zero? && @config["period"].to_i.zero?
+
+      # If the last time a job was executed was more than now + period.seconds ago, reset executed back to 0
+      # This handles executions not in same time frame
+      # Which otherwise would cause throttling to kick in once executed == limit even if the executions were hours apart with a 60 sec period
+      if Time.utc_now.epoch > (Time.epoch(@config["last_executed"].to_i64) + @config["period"].to_i.seconds).epoch
+        @config["executed"] = "0"
+        Redis.instance.store_hash config_q, @config
```

Comment:
I think adding a return false here would make your method read a little easier, even if it is redundant to the line below.

Reply:
:+1: